Firestore/Source/Core/FSTTransaction.mm

```diff
@@ -138,23 +136,29 @@ - (void)writeMutations:(NSArray<FSTMutation *> *)mutations {
  * Returns version of this doc when it was read in this transaction as a precondition, or no
  * precondition if it was not read.
  */
-- (FSTPrecondition *)preconditionForDocumentKey:(FSTDocumentKey *)key {
-  FSTSnapshotVersion *_Nullable snapshotVersion = self.readVersions[key];
-  if (snapshotVersion) {
-    return [FSTPrecondition preconditionWithUpdateTime:snapshotVersion];
-  } else {
+- (FSTPrecondition *)preconditionForDocumentKey:(const DocumentKey &)key {
+  const auto iter = _readVersions.find(key);
+  if (iter == _readVersions.end()) {
```

Comment:
Same as above, this could optionally become:

```
if (readVersions.find(key) == .end()) {
  <same>
} else {
  return [FSTPrecondition preconditionWithUpdateTime:readVersions[key]];
}